File: fe2-android/app/src/test/java/com/github/dedis/popstellar/model/objects/TransactionObjectBuilderTest.java
package com.github.dedis.popstellar.model.objects;

import com.github.dedis.popstellar.model.objects.digitalcash.TransactionObjectBuilder;

import org.junit.Test;

import java.util.ArrayList;

import static org.junit.Assert.assertThrows;

public class TransactionObjectBuilderTest {
  private static final TransactionObjectBuilder builder = new TransactionObjectBuilder();

  @Test
  public void buildWithoutChannelThrowsException() {
    assertThrows(IllegalStateException.class, builder::build);
  }

  @Test
  public void buildWithoutInputsThrowsException() {
    builder.setChannel(Channel.fromString("/root/stuff"));
    assertThrows(IllegalStateException.class, builder::build);
  }

  @Test
  public void buildWithoutOutputsThrowsException() {
    builder.setChannel(Channel.fromString("/root/stuff"));
    builder.setInputs(new ArrayList<>());
    assertThrows(IllegalStateException.class, builder::build);
  }

  @Test
  public void buildWithoutTransactionIdThrowsException() {
    builder.setChannel(Channel.fromString("/root/stuff"));
    builder.setInputs(new ArrayList<>());
    builder.setOutputs(new ArrayList<>());
    assertThrows(IllegalStateException.class, builder::build);
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/TransactionObjectBuilderTest.kt
package com.github.dedis.popstellar.model.objects

import com.github.dedis.popstellar.model.objects.Channel.Companion.fromString
import com.github.dedis.popstellar.model.objects.digitalcash.TransactionObjectBuilder
import org.junit.Assert
import org.junit.Test

class TransactionObjectBuilderTest {
  @Test
  fun buildWithoutChannelThrowsException() {
    Assert.assertThrows(IllegalStateException::class.java) { builder.build() }
  }

  @Test
  fun buildWithoutInputsThrowsException() {
    builder.setChannel(fromString("/root/stuff"))
    Assert.assertThrows(IllegalStateException::class.java) { builder.build() }
  }

  @Test
  fun buildWithoutOutputsThrowsException() {
    builder.setChannel(fromString("/root/stuff"))
    builder.setInputs(ArrayList())
    Assert.assertThrows(IllegalStateException::class.java) { builder.build() }
  }

  @Test
  fun buildWithoutTransactionIdThrowsException() {
    builder.setChannel(fromString("/root/stuff"))
    builder.setInputs(ArrayList())
    builder.setOutputs(ArrayList())
    Assert.assertThrows(IllegalStateException::class.java) { builder.build() }
  }

  companion object {
    private val builder = TransactionObjectBuilder()
  }
}